server.js

```diff
@@ -0,0 +1,306 @@
+                <li>
+                <a href="${lowerCase}">${standardCase.substring(0, subString)} </a>
+                <li>
+                `;
+              }
+
+              let reWrittenHTML = `
+              <!DOCTYPE html>
+              <html lang="en">
+              <head>
+              <meta charset="UTF-8">
+              <title>The Elements</title>
+              <link rel="stylesheet" href="/css/styles.css">
+              </head>
+        <body>
+        <h1>The Elements</h1>
+        <h2>These are all the known elements.</h2>
+        <h3>These are ${knownFiles.length} elements</h3>
+        <ol>
+        ${tags}
+        </ol>
+        </body>
+        </html>`;
+
+              fs.writeFile(`public/index.html`, reWrittenHTML, 'utf8', (err) => {
+                if (err) {
+                  return console.log(err);
+                }
+              });
+            });
+          });
+
+          res.writeHead(200, {
+            'Content-Type': 'application/json',
+          });
+
+          res.end(`{"success" : true}`);
+        }
+      }
+    }
+  }
+  if (req.method === 'PUT') {
+    if (req.headers.authorization) {
+      var encodedString = req.headers.authorization.slice(6);
+      var base64Buffer = new Buffer(encodedString, 'base64');
+      var decodedString = base64Buffer.toString();
+      let username = decodedString.slice(0, decodedString.indexOf(':'));
+      let password = decodedString.slice(decodedString.indexOf(':') + 1);
+
+      if (adminPrivilege.hasOwnProperty(username) && adminPrivilege[username] === password) {
+        let totalData = '';
+
+        req.on('data', (data) => {
+          totalData += data;
+        });
+
+        req.on('end', () => {
+          const parsedData = qs.parse(totalData);
+
+          fs.readdir('./public', (err, files) => {
+            const knownFiles = files.filter(
+              (file) => file !== '.keep' && file !== '.404' && file !== 'css' && file !== 'index.html',
+            );
+
+            if (!knownFiles.includes(`${parsedData.elementName}.html`)) {
+              res.writeHead(500, {
+                'Content-Type': 'application/json',
+              });
+
+              res.end(`{"error"} : ${parsedData.elementName}.html was not found.`);
+            } else {
+              const template = `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>${parsedData.elementName}</title>
+    </head>
+    <body>
+    <div>${parsedData.elementName}</div>
+    <div>${parsedData.elementSymbol}</div>
+    <div>${parsedData.elementAtomicNumber}</div>
+    <div>${parsedData.elementDescription}</div>
+    </body>
+    </html>`;
+
+              fs.writeFile(`./public/${parsedData.elementName}.html`, template, 'utf8', (err) => {
+                if (err) {
+                  throw err;
```

Comment:
You don't handle the err that comes back from fs.writeFile. This should be sending a 500 response back to the client.